
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>contamination &#8212; cfDNApipe 0.1.4 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="createPON" href="createPON.html" />
    <link rel="prev" title="BQSR" href="BQSR.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="contamination">
<h1>contamination<a class="headerlink" href="#contamination" title="Permalink to this headline">¶</a></h1>
<p>This function is used for Calculate the fraction of reads coming from cross-sample contamination using gatk.
Calculates the fraction of reads coming from cross-sample contamination, given results from GetPileupSummaries. The resulting contamination table is used with FilterMutectCalls.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This function is calling gatk ApplyBQSR, please install GATK before using.</p>
<p><a class="reference external" href="https://gatk.broadinstitute.org/hc/en-us/categories/360002310591-Technical-Documentation">gatk official docs</a></p>
<p>McKenna, Aaron, et al. “The Genome Analysis Toolkit: a MapReduce framework for analyzing next-generation DNA sequencing data.” Genome research 20.9 (2010): 1297-1303.</p>
</div>
<div class="section" id="parameters">
<h2>Parameters<a class="headerlink" href="#parameters" title="Permalink to this headline">¶</a></h2>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">contamination</span><span class="p">(</span><span class="n">bamInput</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">contaminationInput</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span>
              <span class="n">outputdir</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">stepNum</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">threads</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
              <span class="n">upstream</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
<ul class="simple">
<li><p>bamInput: list, bam files, just as Output for next step analysis.</p></li>
<li><p>contaminationInput: str, Tabulates pileup metrics files from getPileup.</p></li>
<li><p>outputdir: str, output result folder, None means the same folder as input files.</p></li>
<li><p>threads: int, how many thread to use.</p></li>
<li><p>stepNum: int or str, step flag for folder name.</p></li>
<li><p>upstream: upstream output results, used for pipeline, just can be getPileup. This parameter can be True, which means a new pipeline start.</p></li>
<li><p>verbose: bool, True means print all stdout, but will be slow; False means black stdout verbose, much faster.</p></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>We recommend using this function in SNV detection.
For a detailed tutorial, please see <a class="reference external" href="https://honchkrow.github.io/cfDNApipe/#section-6-additional-function-wgs-snvindel-analysis">SNV detection tutorial</a>.</p>
</div>
<p>Example usage:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">cfDNApipe</span> <span class="kn">import</span> <span class="o">*</span>
<span class="kn">import</span> <span class="nn">glob</span>

<span class="n">pipeConfigure</span><span class="p">(</span>
    <span class="n">threads</span><span class="o">=</span><span class="mi">60</span><span class="p">,</span>
    <span class="n">genome</span><span class="o">=</span><span class="s2">&quot;hg19&quot;</span><span class="p">,</span>
    <span class="n">refdir</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;path_to_reference/hg19&quot;</span><span class="p">,</span>
    <span class="n">outdir</span><span class="o">=</span><span class="sa">r</span><span class="s2">&quot;path_to_output/snv_output&quot;</span><span class="p">,</span>
    <span class="n">data</span><span class="o">=</span><span class="s2">&quot;WGS&quot;</span><span class="p">,</span>
    <span class="nb">type</span><span class="o">=</span><span class="s2">&quot;paired&quot;</span><span class="p">,</span>
    <span class="n">build</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span>
    <span class="n">JavaMem</span><span class="o">=</span><span class="s2">&quot;10G&quot;</span><span class="p">,</span>
<span class="p">)</span>

<span class="c1"># just set build=True to finish all the works</span>
<span class="n">Configure</span><span class="o">.</span><span class="n">snvRefCheck</span><span class="p">(</span><span class="n">folder</span><span class="o">=</span><span class="s2">&quot;path_to_reference/hg19/hg19_snv&quot;</span><span class="p">,</span> <span class="n">build</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="c1"># see all the SNV related files</span>
<span class="n">Configure</span><span class="o">.</span><span class="n">getConfig</span><span class="p">(</span><span class="s2">&quot;snv.folder&quot;</span><span class="p">)</span>
<span class="n">Configure</span><span class="o">.</span><span class="n">getConfig</span><span class="p">(</span><span class="s2">&quot;snv.ref&quot;</span><span class="p">)</span>

<span class="c1"># indexed bam files after remove duplicates</span>
<span class="n">bams</span> <span class="o">=</span> <span class="n">glob</span><span class="o">.</span><span class="n">glob</span><span class="p">(</span><span class="s2">&quot;path_to_samples/*.bam&quot;</span><span class="p">)</span>

<span class="n">res1</span> <span class="o">=</span> <span class="n">addRG</span><span class="p">(</span><span class="n">bamInput</span><span class="o">=</span><span class="n">bams</span><span class="p">,</span> <span class="n">upstream</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>

<span class="n">res2</span> <span class="o">=</span> <span class="n">BaseRecalibrator</span><span class="p">(</span>
    <span class="n">upstream</span><span class="o">=</span><span class="n">res1</span><span class="p">,</span> <span class="n">knownSitesDir</span><span class="o">=</span><span class="n">Configure</span><span class="o">.</span><span class="n">getConfig</span><span class="p">(</span><span class="s2">&quot;snv.folder&quot;</span><span class="p">)</span>
<span class="p">)</span>
<span class="n">res3</span> <span class="o">=</span> <span class="n">BQSR</span><span class="p">(</span><span class="n">upstream</span><span class="o">=</span><span class="n">res2</span><span class="p">)</span>

<span class="n">res4</span> <span class="o">=</span> <span class="n">getPileup</span><span class="p">(</span>
    <span class="n">upstream</span><span class="o">=</span><span class="n">res3</span><span class="p">,</span>
    <span class="n">biallelicvcfInput</span><span class="o">=</span><span class="n">Configure</span><span class="o">.</span><span class="n">getConfig</span><span class="p">(</span><span class="s1">&#39;snv.ref&#39;</span><span class="p">)[</span><span class="s2">&quot;7&quot;</span><span class="p">],</span>
<span class="p">)</span>

<span class="n">res5</span> <span class="o">=</span> <span class="n">contamination</span><span class="p">(</span><span class="n">upstream</span><span class="o">=</span><span class="n">res4</span><span class="p">)</span>

<span class="c1"># In this step, files are split to chromatin</span>
<span class="n">res6</span> <span class="o">=</span> <span class="n">mutect2t</span><span class="p">(</span>
    <span class="n">caseupstream</span> <span class="o">=</span> <span class="n">res5</span><span class="p">,</span>
    <span class="n">vcfInput</span><span class="o">=</span><span class="n">Configure</span><span class="o">.</span><span class="n">getConfig</span><span class="p">(</span><span class="s1">&#39;snv.ref&#39;</span><span class="p">)[</span><span class="s2">&quot;6&quot;</span><span class="p">],</span>
    <span class="n">ponbedInput</span><span class="o">=</span><span class="n">Configure</span><span class="o">.</span><span class="n">getConfig</span><span class="p">(</span><span class="s1">&#39;snv.ref&#39;</span><span class="p">)[</span><span class="s2">&quot;8&quot;</span><span class="p">],</span>
<span class="p">)</span>

<span class="c1">#</span>
<span class="n">res7</span> <span class="o">=</span> <span class="n">filterMutectCalls</span><span class="p">(</span><span class="n">upstream</span><span class="o">=</span><span class="n">res6</span><span class="p">)</span>

<span class="c1"># put all chromatin files together</span>
<span class="n">res8</span> <span class="o">=</span> <span class="n">gatherVCF</span><span class="p">(</span><span class="n">upstream</span><span class="o">=</span><span class="n">res7</span><span class="p">)</span>

<span class="c1"># split somatic mutations</span>
<span class="n">res9</span> <span class="o">=</span> <span class="n">bcftoolsVCF</span><span class="p">(</span><span class="n">upstream</span><span class="o">=</span><span class="n">res8</span><span class="p">,</span> <span class="n">stepNum</span><span class="o">=</span><span class="s2">&quot;somatic&quot;</span><span class="p">)</span>

<span class="c1"># split germline mutations</span>
<span class="n">res10</span> <span class="o">=</span> <span class="n">bcftoolsVCF</span><span class="p">(</span>
    <span class="n">upstream</span><span class="o">=</span><span class="n">res8</span><span class="p">,</span> <span class="n">other_params</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;-f&quot;</span><span class="p">:</span> <span class="s2">&quot;&#39;germline&#39;&quot;</span><span class="p">},</span> <span class="n">suffix</span><span class="o">=</span><span class="s2">&quot;germline&quot;</span><span class="p">,</span> <span class="n">stepNum</span><span class="o">=</span><span class="s2">&quot;germline&quot;</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">cfDNApipe</a></h1>








<h3>Navigation</h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Up-Down-Stream-Flowchart.html">Up Down Stream Flowchart</a></li>
<li class="toctree-l1"><a class="reference internal" href="Basic-Data-Processing.html">Basic Data Processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="Quality-Control.html">Quality Control</a></li>
<li class="toctree-l1"><a class="reference internal" href="Methylation-Analysis.html">Methylation Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="Large-Scale-CNV.html">Large Scale CNV</a></li>
<li class="toctree-l1"><a class="reference internal" href="Bin-Scale-CNV.html">Bin Scale CNV</a></li>
<li class="toctree-l1"><a class="reference internal" href="Fragmentomics.html">Fragmentomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="Virus-Detection.html">Virus Detection</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="SNV-Detection.html">SNV Detection</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="addRG.html">addRG</a></li>
<li class="toctree-l2"><a class="reference internal" href="BaseRecalibrator.html">BaseRecalibrator</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcftoolsVCF.html">bcftoolsVCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="BQSR.html">BQSR</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">contamination</a></li>
<li class="toctree-l2"><a class="reference internal" href="createPON.html">createPON</a></li>
<li class="toctree-l2"><a class="reference internal" href="dbimport.html">dbimport</a></li>
<li class="toctree-l2"><a class="reference internal" href="filterMutectCalls.html">filterMutectCalls</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatherVCF.html">gatherVCF</a></li>
<li class="toctree-l2"><a class="reference internal" href="getPileup.html">getPileup</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2n.html">mutect2n</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutect2t.html">mutect2t</a></li>
<li class="toctree-l2"><a class="reference internal" href="SNV-Detection.html#snv-example-usage">SNV Example usage</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
  <li><a href="SNV-Detection.html">SNV Detection</a><ul>
      <li>Previous: <a href="BQSR.html" title="previous chapter">BQSR</a></li>
      <li>Next: <a href="createPON.html" title="next chapter">createPON</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2021, Wei Zhang.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 3.2.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/contamination.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>